### PR TITLE
[#5520] Correctly include all PoolSubpage metrics

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

Because of a bug we missed to include the first PoolSubpage when collection metrics.

Modifications:

- Correctly include all subpages
- Add unit test

Result:

Correctly include all subpages

### CODE REVIEW

**Comment by normanmaurer** [comment]
@Scottmitch @nmittler @trustin PTAL

File: buffer/src/main/java/io/netty/buffer/PoolArena.java [hunk 1]
```diff
@@ -589,36 +589,26 @@ public synchronized String toString() {
...
            .append("small subpages:");
-        for (int i = 1; i < smallSubpagePools.length; i ++) {
-            PoolSubpage<T> head = smallSubpagePools[i];
+        appendPoolSubPages(buf, smallSubpagePools);
+        buf.append(StringUtil.NEWLINE);
+
+        return buf.toString();
+    }
+
+    private static void appendPoolSubPages(StringBuilder buf, PoolSubpage<?>[] subpages) {
```
Comment (netkins) [inline buffer/src/main/java/io/netty/buffer/PoolArena.java]: MINOR Consider using varargs for methods or constructors which take an array the last parameter. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseVarargs)

Comment (normanmaurer) [inline buffer/src/main/java/io/netty/buffer/PoolArena.java]: I think this is ok and also consistent with the rest of this class.

**Comment by nmittler** [comment]
LGTM

File: buffer/src/test/java/io/netty/buffer/PooledByteBufAllocatorTest.java [hunk 2]
```diff
@@ -95,6 +95,26 @@ private static void assertPoolChunkListMetric(PoolChunkListMetric m, int min, in
...
+    }
+
+    @Test
+    public void testTinySubpageMetric() {
+        PooledByteBufAllocator allocator = new PooledByteBufAllocator(true, 1, 1, 8192, 11, 0, 0, 0);
+        ByteBuf buffer = allocator.heapBuffer(1);
+        PoolArenaMetric metric = allocator.heapArenas().get(0);
+        PoolSubpageMetric subpageMetric = metric.tinySubpages().get(0);
+        assertEquals(1, subpageMetric.maxNumElements() - subpageMetric.numAvailable());
+        buffer.release();
```
Comment (Scottmitch) [inline buffer/src/test/java/io/netty/buffer/PooledByteBufAllocatorTest.java]: nit: put this in a finally block?

**Comment by Scottmitch** [comment]
1 comment and lgtm

**Comment by normanmaurer** [comment]
Cherry-picked into 4.0 (196540c) and 4.1 (67e54873d0d45d6fa94cbcd2b646a154dcc69ec7)

**Pull request closed without merging**